### PR TITLE
Update the `Message` documentation.

### PR DESCRIPTION
**Pull request by dpassage**
Updates #236 given the new API layout. `Message` is now fully documented.

### CODE REVIEW

File: Sources/SwiftProtobuf/Google_Protobuf_Any.swift [hunk 1]
```diff
@@ -83,6 +83,9 @@ public extension Message {
   ///
   /// See `Google_Protobuf_Any.unpackTo()` for more discussion.
   ///
+  /// - Parameter unpackingAny: the message to decode.
+  /// - Throws: an instance of `AnyUnpackError`, `JSONDecodingError`, or
+  ///   `BinaryDecodingError` on failure.
   public init(unpackingAny: Google_Protobuf_Any) throws {
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Google_Protobuf_Any.swift]: fyi - https://github.com/apple/swift-protobuf/pull/356 could end up changing this.

File: Sources/SwiftProtobuf/Message.swift [hunk 2]
```diff
@@ -67,6 +81,12 @@ public protocol Message: CustomDebugStringConvertible {
   /// semantics for messages with heap storage; it should only be called on
   /// newly-created messages or messages where the storage has been ensured
   /// unique.
+  ///
+  /// - Parameters:
+  ///   - decoder: a `Decoder`; the `Message` will call the method
+  ///     corresponding to the type of this field.
+  ///   - protoFieldNumber: the number of the field to decode.
+  /// - Throws: an error on failure or type mismatch.
   mutating func decodeField<D: Decoder>(decoder: inout D, fieldNumber: Int) throws
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: I believe this is meant to be internal to the library, and I thought @allevato was hoping to maybe even remove it.

File: Sources/SwiftProtobuf/Message.swift [hunk 3]
```diff
@@ -76,18 +96,19 @@ public protocol Message: CustomDebugStringConvertible {
   /// then invokes `decodeField` above.
   mutating func decodeMessage<D: Decoder>(decoder: inout D) throws
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: Same, internal to the library?

File: Sources/SwiftProtobuf/Message.swift [hunk 4]
```diff
@@ -96,15 +117,22 @@ public protocol Message: CustomDebugStringConvertible {
   // JSON encoding/decoding support
   //
 
-  /// Returns a JSON-coded representation of this object as a String.
+  /// Returns a JSON-encoded representation of this object as a `String`.
   func jsonString() throws -> String
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: @tbkka didn't this move out with the other json support?

File: Sources/SwiftProtobuf/Message.swift [hunk 5]
```diff
@@ -142,7 +184,12 @@ public extension Message {
     return result
   }
 
+  /// The literal `type.googleapis.com`; may be overridden if your
+  /// message's type should be encoded differently.
   static var anyTypePrefix: String { return "type.googleapis.com" }
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: These two may also be going away with that PR.

File: Sources/SwiftProtobuf/Message.swift (hunk 4 above)
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: I guess not, should we move the overrides of this into _CustomJSONCodable and use that same pattern?

File: Sources/SwiftProtobuf/Message.swift (hunk 2 above)
Comment (allevato) [inline Sources/SwiftProtobuf/Message.swift]: Correct—this method is mostly only internal to the library, and someone should only call it directly if they were implementing their own `Decoder`. The documentation should reflect that.

File: Sources/SwiftProtobuf/Message.swift (hunk 3 above)
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: Scratch that, I think this was the one we wanted to keep public, but decodeField was the one we were sorta hopping to kill/making internal/etc.

Comment (allevato) [inline Sources/SwiftProtobuf/Message.swift]: Yes, same as above.

File: Sources/SwiftProtobuf/Message.swift (hunk 2 above)
Comment (dpassage) [inline Sources/SwiftProtobuf/Message.swift]: check, adding text to that effect.

File: Sources/SwiftProtobuf/Google_Protobuf_Any.swift (hunk 1 above)
Comment (dpassage) [inline Sources/SwiftProtobuf/Google_Protobuf_Any.swift]: Should we leave its documentation alone, then?

File: Sources/SwiftProtobuf/Message.swift (hunk 5 above)
Comment (dpassage) [inline Sources/SwiftProtobuf/Message.swift]: Should I remove the docs, then?

File: Sources/SwiftProtobuf/Google_Protobuf_Any.swift (hunk 1 above)
Comment (thomasvl) [inline Sources/SwiftProtobuf/Google_Protobuf_Any.swift]: You can update it, and try to merge up/update my PR after it lands.

File: Sources/SwiftProtobuf/Message.swift (hunk 4 above)
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: fyi - #362 will move this out.

File: Sources/SwiftProtobuf/Message.swift (hunk 5 above)
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: Eh, I guess keep them so @tbkka can agree with my PR.

File: Sources/SwiftProtobuf/Message.swift [hunk 6]
```diff
@@ -54,10 +68,12 @@ public protocol Message: CustomDebugStringConvertible {
   //
 
   /// Decode a field identified by a field number (as given in the .proto file).
-  /// The Message will call the Decoder method corresponding
+  /// The `Message` will call the Decoder method corresponding
   /// to the declared type of the field.
   ///
-  /// This is the core method used by the deserialization machinery.
+  /// This is the core method used by the deserialization machinery. It is
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/Message.swift]: I think you want this text also on decodeMessage

File: Sources/SwiftProtobuf/Message.swift (hunk 4 above)
Comment (tbkka) [inline Sources/SwiftProtobuf/Message.swift]: #362 moved `func jsonString()` out of the Message protocol, but it is still part of the public API and still needs to be documented in `JSONTypeAdditions.swift`.

File: Sources/SwiftProtobuf/JSONTypeAdditions.swift [hunk 7]
```diff
@@ -15,9 +15,9 @@
...
 import Foundation
 
-///
-/// Messages
-///
+//
+// Messages
+//
 public extension Message {
     func jsonString() throws -> String {
```
Comment (tbkka) [inline Sources/SwiftProtobuf/JSONTypeAdditions.swift]: This should be public and should be documented.

**Comment by dpassage** [comment]
Thanks, @tbkka !

**Pull request merged**